### PR TITLE
WIP - LinSolveCG erroring due to mismatched kwargs

### PR DESCRIPTION
**Pull request by vpuri3**
MWE : https://github.com/SciML/OrdinaryDiffEq.jl/issues/1526

### CODE REVIEW

**Comment by vpuri3** [comment]
should also use `get` functions as in https://github.com/SciML/DiffEqBase.jl/issues/366

**Comment by vpuri3** [comment]
other `IterativeSolvers.jl` related issues: 
https://github.com/SciML/DiffEqBase.jl/issues/365
https://github.com/SciML/DiffEqBase.jl/issues/366
https://github.com/SciML/OrdinaryDiffEq.jl/issues/1430
https://github.com/SciML/OrdinaryDiffEq.jl/issues/953

File: src/linear_nonlinear.jl [hunk 1]
```diff
@@ -220,17 +220,30 @@ function (f::LinSolveIterativeSolvers)(x,A,b,update_matrix=false; Pl=nothing, Pr
     Pr = ComposePreconditioner(f.Pr, Pr, false)
 
     reltol = checkreltol(reltol)
-    f.iterable = f.generate_iterator(x,A,b,f.args...;
-                                     initially_zero=true,restart=5,
-                                     maxiter=5,abstol=1e-16,reltol=reltol,
-                                     Pl=Pl,Pr=Pr,
-                                     kwargs...,f.kwargs...)
+#   @show f.generate_iterator
+    if f.generate_iterator == IterativeSolvers.cg_iterator!
```
Comment (ChrisRackauckas) [inline src/linear_nonlinear.jl:224]:
```suggestion
    if f.generate_iterator === IterativeSolvers.cg_iterator!

```

**Comment by ChrisRackauckas** [comment]
I wouldn't put too much time into this as it's being deprecated soon for LinearSolve.jl.

**Comment by vpuri3** [comment]
closing since this Krylov wrappers have been tested in https://github.com/SciML/LinearSolve.jl

**Pull request closed without merging**